src/jvm/main/org/jetbrains/kotlinx/lincheck/CTestStructure.java

```diff
@@ -199,7 +246,14 @@ private static ParameterGenerator<?> getOrCreateGenerator(Method m,
         if (!paramAnn.name().isEmpty())
             return checkAndGetNamedGenerator(namedGens, paramAnn.name());
         // Otherwise create new parameter generator
-        return createGenerator(paramAnn, randomProvider);
+        if (p.getType().isEnum()) {
```

Comment:
Would moving this logic to "createGenerator" improve the code clarity?

Reply:
I think no, as we use different conditions to call first or second factory method in another places.